clients/src/main/java/org/apache/kafka/common/network/Selector.java

```diff
@@ -106,7 +105,7 @@
     private boolean outOfMemory;
     private final List<Send> completedSends;
     private final List<NetworkReceive> completedReceives;
-    private final Map<KafkaChannel, Deque<NetworkReceive>> stagedReceives;
+    private final Set<KafkaChannel> channelsWithCompletedReceives;
```

Comment:
@rajinisivaram did we consider changing `completedReceives` to be a `LinkedHashMap` from `channel` to `NetworkReceive`?

Reply:
@ijuma Thanks for the review, updated.